### PR TITLE
[ci][terraform] tags with metadata

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Standardise the below tags/labels in the terraform resources for AWS/GCP:

- `environment` => static value
- `repo`.              => static value
- `branch`           => dynamic value
- `build`              => dynamic value 
- `created_date` => dynamic value

Bump the aws terraform version to https://github.com/hashicorp/terraform-provider-aws/blob/v3.33.0/CHANGELOG.md 

In addition, I refactored the cloud stage to use one node rather than two, since it created the context in one node and then run the tests in another node

## Why is it important?

Help with tearing down any of the ephemeral resources which were not successfully removed as part of the system tests.

## Implementation details

* Those tag/labels are lowercase based to be [GCP](https://cloud.google.com/resource-manager/docs/creating-managing-labels#requirements)/[AWS](https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html#tag-conventions) compliance
* `environment=CI` will allow to filter all those cloud resources which were created in the CI
* While `created_date` will help to filter those which were older than X days or X hours, since it's Unix epoch time based.
* `branch` and `build` will help to identify what build in the CI and what PR, Branch or Tag created those resources.

## Further details

> BUILD_ID
The current build ID, identical to BUILD_NUMBER for builds created in 1.597+, but a YYYY-MM-DD_hh-mm-ss timestamp for older builds.

> BRANCH_NAME
For a multibranch project, this will be set to the name of the branch being built, for example in case you wish to deploy to production from master but not from feature branches; if corresponding to some kind of change request, the name is generally arbitrary (refer to CHANGE_ID and CHANGE_TARGET).

## Test

image
https://beats-ci.elastic.co/job/Beats/job/beats/job/PR-31355/9/artifact/x-pack/filebeat/input/awss3/_meta/terraform/terraform.tfstate/*view*/

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -930,7 +930,11 @@ def startCloudTestEnv(Map args = [:]) {
 def terraformApply(String directory) {
   terraformInit(directory)
   dir(directory) {
-    sh(label: "Terraform Apply on ${directory}", script: "terraform apply -auto-approve")
+    withEnv(["TF_VAR_BUILD_ID=${BUILD_ID}",
+             "TF_VAR_BRANCH_NAME=${BRANCH_NAME}",
+             "TF_VAR_CREATED_DATE=${current.getRawBuild().getTimestampString2()}"]) {
```
Comment (v1v) [inline Jenkinsfile:935]:
```suggestion
             "TF_VAR_CREATED_DATE=${currentBuild.getRawBuild().getTimestampString2()}"]) {
```

File: x-pack/metricbeat/module/aws/terraform.tf [hunk 2]
```diff
@@ -1,5 +1,14 @@
 provider "aws" {
-  version = "~> 2.58"
+  version = "~> 3.33"
```
Comment (v1v) [inline x-pack/metricbeat/module/aws/terraform.tf:2]: Forced to bump the version to be able to use the `default_tags`:
- https://github.com/hashicorp/terraform-provider-aws/blob/v3.33.0/CHANGELOG.md

**Review by kuisathaverat — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Review by kvch — APPROVED** [review]

**Pull request merged**